pkg/operator/reconciler/run/instrumentation.go

```diff
@@ -10,34 +10,68 @@ const (
 	metricWorkflowRunLogUploadDuration = "workflow_run_log_upload_duration"
 )
 
+type trackDurationOptions struct {
+	accountID string
+}
+
+type trackDurationOption interface {
+	apply(*trackDurationOptions)
+}
+
+type trackDurationOptionFunc struct {
+	fn func(*trackDurationOptions)
+}
+
+func (o trackDurationOptionFunc) apply(opts *trackDurationOptions) {
+	o.fn(opts)
+}
+
+func withAccountIDTrackDurationOption(accountID string) trackDurationOptionFunc {
+	return trackDurationOptionFunc{
+		fn: func(opts *trackDurationOptions) {
+			opts.accountID = accountID
+		},
+	}
+}
+
 type controllerObservations struct {
 	mets *metrics.Metrics
 }
 
-func (c *controllerObservations) trackDurationWithOutcome(metric string, fn func() error) error {
+func (c *controllerObservations) trackDurationWithOutcome(metric string, fn func() error, opts ...trackDurationOption) error {
+	o := trackDurationOptions{
+		accountID: "none",
```

Comment:
Would it be better to just leave this blank/empty string/omit it from the observation altogether? Or does that not work correctly with the way labels are used for grouping?

Reply:
So it looks like omiting the label, once it's registered for the metric, is an error and blank values have been slightly problematic in my testing. I think I'm going to leave this value "none" for now. I'd like to circle back around later and figure out a default across the board for unknown values as we are a bit splitbrained right now.